models/keyword.go

```diff
@@ -0,0 +1,65 @@
+package models
+
+import (
+	"database/sql/driver"
+	"errors"
+
+	"github.com/beego/beego/v2/client/orm"
+)
+
+type KeywordStatus string
+
+const (
+	Failed     KeywordStatus = "failed"
+	Pending    KeywordStatus = "pending"
+	Processed  KeywordStatus = "processed"
+	Processing KeywordStatus = "processing"
+
+	InvalidKeywordStatusErr = "invalid keyword status"
+)
+
+func (k KeywordStatus) Value() (driver.Value, error) {
+	switch k {
+	case Pending, Processing, Processed, Failed:
+		return string(k), nil
+	}
+	return nil, errors.New(InvalidKeywordStatusErr)
+}
+
+type Keyword struct {
+	Base
+
+	Keyword string
+	Status  KeywordStatus `orm:"type(KeywordStatus);default(pending)"`
+
+	LinksCount           int
+	NonAdwordLinks       string `orm:"type(json)"`
+	NonAdwordLinksCount  int
+	AdwordLinks          string `orm:"type(json)"`
+	AdwordLinksCount     int
+	ShopAdwordLinks      string `orm:"type(json)"`
+	ShopAdwordLinksCount int
+	HtmlCode             string
+
+	User *User `orm:"rel(fk)"`
+}
+
+func init() {
+	orm.RegisterModel(new(Keyword))
+}
+
+// CreateKeyword insert a new Keyword into database and returns last inserted Id on success.
```

Comment:
MInor English syntax issue (same issue beliow for `update`)
```suggestion 
// CreateKeyword inserts a new Keyword into database and returns last inserted Id on success.
```